### ISSUE TITLE
keyboard_enter_text times out typing long text

### ISSUE THREAD

**Issue by jmoody** [calabash/calabash-ios#1158]
Thanks @JoeSSS for reporting.

**Comment by stopiccot** [comment]
Not sure if it's the same bug but we are experiencing keyboard_enter_text timeouts too. But in our case it's nothing to deal with long text. Here is output:
```
And I enter "name.surname@email.com" into text field number 1
  execution expired (HTTPClient::ReceiveTimeoutError)
```
As you can see it's not a long text. On the simulator side correct text field is focused but no text is inputed at all. I must note that this bug occurs from time to time and not on all test machines.

And a related question. I would like to debug\read DeviceAgent's code to figure out what is the source of problem but I can't find any repos. Is it closed-source?

**Comment by jmoody** [comment]
> DeviceAgent

Closed source with plans to open source it in the near future.

**Comment by jmoody** [comment]
We have reproduced this problem and are trying to find a solution.

0.20.3 will contain some patches that _might_ help.

This is a very high priority issue.

ATTN: @jescriba

**Comment by jmoody** [comment]
Cannot reproduce with DeviceAgent 1.0.3 (not released yet).

**Comment by jmoody** [comment]
@stopiccot Can you reproduce with run_loop 2.2.4?

**Comment by bhaskarsr** [comment]
@jmoody Yes ! I am facing this issue with run_loop 2.2.4 as well. keyboard_enter_text frequently does not input the text into the text field. Before failing, it just enters first character and then fails with the error 

"Expected JSON response with no error, but found
Timed out waiting for key event to complete
(RunLoop::DeviceAgent::Client::HTTPError)"

**Comment by yeshwanthpulagam** [comment]
@bhaskarsr The issue resolved for me when i changed the calabash-ios server to 0.20.4,might be that the present version is 0.20.3
https://github.com/calabash/calabash-ios/wiki/Updating-your-Calabash-iOS-version

**Comment by Simran-Jit-Singh** [comment]
I am still facing the same issue
`Expected JSON response with no error, but found

      Timed out waiting for key event to complete

       (RunLoop::DeviceAgent::Client::HTTPError)
`
The first character is read properly but fails for next. Characters are Integer and float.
run loop version 2.4.1
calabash version 0.20.0
